examples/multiple_viewer_widget.py

```diff
@@ -0,0 +1,448 @@
+    for event_name, event_emitter in layer.events.emitters.items():
+        if isinstance(event_emitter, WarningEmitter):
+            continue
+        if event_name in ("thumbnail", "name"):
+            continue
+        if (
+            isinstance(getattr(klass, event_name, None), property)
+            and getattr(klass, event_name).fset is not None
+        ):
+            res.append(event_name)
+    return res
+
+
+def center_cross_on_mouse(
+    viewer_model: napari.components.viewer_model.ViewerModel,
+):
+    """move the cross to the mouse position"""
+
+    if not getattr(viewer_model, "_mouse_over_canvas", True):
+        # There is no way for napari 0.4.15 to check if mouse is over sending canvas.
+        show_info(
+            "Mouse is not over the canvas. You may need to click on the canvas."
+        )
+        return
+
+    viewer_model.dims.current_step = tuple(
+        np.round(
+            [
+                max(min_, min(p, max_)) / step
+                for p, (min_, max_, step) in zip(
+                    viewer_model.cursor.position, viewer_model.dims.range
+                )
+            ]
+        ).astype(int)
+    )
+
+
+action_manager.register_action(
+    name='napari:move_point',
+    command=center_cross_on_mouse,
+    description='Move dims point to mouse position',
+    keymapprovider=ViewerModel,
+)
+
+action_manager.bind_shortcut('napari:move_point', 'C')
+
+
+class own_partial:
+    """
+    Workaround for deepcopy not copying partial functions
+    (Qt widgets are not serializable)
+    """
+
+    def __init__(self, func, *args, **kwargs):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+    def __call__(self, *args, **kwargs):
+        return self.func(*(self.args + args), **{**self.kwargs, **kwargs})
+
+    def __deepcopy__(self, memodict={}):
+        return own_partial(
+            self.func,
+            *deepcopy(self.args, memodict),
+            **deepcopy(self.kwargs, memodict),
+        )
+
+
+class QtViewerWrap(QtViewer):
+    def __init__(self, main_viewer, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.main_viewer = main_viewer
+
+    def _qt_open(
+        self,
+        filenames: list,
+        stack: bool,
+        plugin: str = None,
+        layer_type: str = None,
+        **kwargs,
+    ):
+        """for drag and drop open files"""
+        self.main_viewer.window._qt_viewer._qt_open(
+            filenames, stack, plugin, layer_type, **kwargs
+        )
+
+
+class CrossWidget(QCheckBox):
+    """
+    Widget to control the cross layer. because of the performance reason
+    the cross update is throttled
+    """
+
+    def __init__(self, viewer: napari.Viewer):
+        super().__init__("Add cross layer")
+        self.viewer = viewer
+        self.setChecked(False)
+        self.stateChanged.connect(self._update_cross_visibility)
+        self.layer = None  # Vectors(name=".cross", ndim=self.viewer.dims.ndim)
```

Comment:
I'm not sure how to read the implementation comment to the right of this. Either add some explanation or remove it.